eva-tools/src/main/java/embl/ebi/variation/eva/vcfdump/VariantExporter.java

```diff
@@ -65,43 +64,91 @@
-        regionSequence = null;
+        variantToVariantContextConverter.cleanCachedRegionSequence();
 
         while (iterator.hasNext()) {
             Variant variant = iterator.next();
             if (region.contains(variant.getChromosome(), variant.getStart())) {
                 try {
-                    Map<String, VariantContext> variantContexts = convertBiodataVariantToVariantContext(variant, studyIds, region);
-                    for (Map.Entry<String, VariantContext> variantContext : variantContexts.entrySet()) {
-                        variantsToExportByStudy.putIfAbsent(variantContext.getKey(), new ArrayList<>());
-                        variantsToExportByStudy.get(variantContext.getKey()).add(variantContext.getValue());
-                    }
+                    VariantContext variantContext = variantToVariantContextConverter.transform(variant, region);
+                    variantsToExport.add(variantContext);
                 } catch (Exception e) {
                     logger.warn("Variant {}:{}:{}>{} dump failed: {}", variant.getChromosome(), variant.getStart(), variant.getReference(),
                             variant.getAlternate(), e.getMessage());
                     failedVariants++;
                 }
             }
         }
-        return variantsToExportByStudy;
+        return variantsToExport;
     }
 
-    public Map<String, VariantSource> getSources(VariantSourceDBAdaptor sourceDBAdaptor, List<String> studyIds) {
+    public Map<String, VariantSource> getSources(VariantSourceDBAdaptor sourceDBAdaptor, List<String> studyIds) throws IllegalArgumentException {
         Map<String, VariantSource> sources = new TreeMap<>();
         List<VariantSource> sourcesList = sourceDBAdaptor.getAllSourcesByStudyIds(studyIds, new QueryOptions()).getResult();
+        checkIfTereAreSourceForEveryStudy(studyIds, sourcesList);
+        Map<String, Map<String, String>> studiesSampleNamesMapping = checkIfConflictsInSampleNames(sourcesList);
+        variantToVariantContextConverter = new BiodataVariantToVariantContextConverter(sourcesList, cellbaseClient, studiesSampleNamesMapping);
         for (VariantSource variantSource : sourcesList) {
             sources.put(variantSource.getStudyId(), variantSource);
         }
         return sources;
     }
 
+    private void checkIfTereAreSourceForEveryStudy(List<String> studyIds, List<VariantSource> sourcesList) throws IllegalArgumentException {
+        List<String> missingStudies =
+                studyIds.stream().filter(study -> sourcesList.stream().noneMatch(source -> source.getStudyId().equals(study))).collect(Collectors.toList());
+        if (!missingStudies.isEmpty()) {
+            throw new IllegalArgumentException("Study(ies) " + String.join(", ", missingStudies) + " not found");
+        }
+    }
+
+    public Map<String, Map<String, String>> checkIfConflictsInSampleNames(Collection<VariantSource> sources) {
```

Comment:
I think a name like "createNonConflictingSampleNames" or similar could explain better this method's final goal.